receiver/mongodbatlasreceiver/config.go

```diff
@@ -129,11 +132,22 @@ var (
 
 	// Access Logs Errors
 	errMaxPageSize = errors.New("the maximum value for 'page_size' is 20000")
+
+	// Config Errors
+	errConfigEmptyEndpoint = errors.New("baseurl must not be empty")
 )
 
 func (c *Config) Validate() error {
 	var errs error
 
+	if c.BaseURL == "" {
+		c.BaseURL = mongodbatlas.CloudURL
+	}
```

Comment:
```suggestion
```

Ideally we shouldn't be setting anything in the validate method and should rely on `createDefaultConfig()` From an API design principle we shouldn't be modifying the struct within a validate call. I know it might seem tedious but in our `config_test.go` cases we should just be setting the BaseURL as mongodbatlas.CloudURL

i.e.

```go
		{
			name: "invalid Metrics Config",
			input: Config{
				BaseURL: mongodbatlas.CloudURL,
				Projects: []*ProjectConfig{
					{
						Name:            "Project1",
						IncludeClusters: []string{"Cluster1"},
						ExcludeClusters: []string{"Cluster2"},
					},
				},
				ControllerConfig: scraperhelper.NewDefaultControllerConfig(),
			},
			expectedErr: errClusterConfig.Error(),
		},
```